zeus/tests/test_poll_views.py

```diff
@@ -0,0 +1,164 @@
+            "\ \  \ \  \ \  \/  /_\n"
+            " \ \  \ \  \ \   ___ \\\n"
+            "  \ \  \_\  \ \  \\\ \ \\ \n"
+            "   \ \_______\ \__\\\ \_\\\n"
+            "    \|_______|\|__| \|_|\n"
+            )
+
+    def verbose(self, message):
+        if self.local_verbose:
+            print message
+
+
+    def get_institution(self, **kwargs):
+        '''
+        Create, or retrieve an instituion. For testing purposes we only need one, so
+        this instance is effectively a singleton.
+        '''
+        institution, _ = Institution.objects.get_or_create(**kwargs)
+        return institution
+
+    def get_election(self, **kwargs):
+        self.c.post(self.locations['login'], self.login_data)
+        institution = self.get_institution()
+        election, _ = Election.objects.get_or_create(institution=institution, **kwargs)
+        return election
+
+    def get_messages_from_response(self, response):
+        messages = []
+        for item in response.context['messages']:
+            messages.append(unicode(item))
+        return messages
+
+    def create_poll(self, election, **kwargs):
+        self.c.post(self.locations['login'], self.login_data)
+        poll, _ = Poll.objects.get_or_create(election=election, **kwargs)
+        self.p_uuids = []
+        for poll in election.polls.all():
+            self.p_uuids.append(poll.uuid)
+        return poll
+
+    def get_voters_file(self):
+        counter = 0
+        voter_files = {}
+        for p_uuid in self.p_uuids:
+            fname = '/tmp/random_voters%s.csv' % counter
+            voter_files[p_uuid] = fname
+            fp = file(fname, 'w')
+            for i in range(1, self.voters_num+1):
+                voter = "%s,voter%s@mail.com,test_name%s,test_surname%s\n" \
+                    % (i, i, i, i)
+                fp.write(voter)
+            fp.close()
+            counter += 1
+        self.verbose('+ Voters file created')
+        return voter_files
+
+    def submit_voters_file(self):
+        voter_files = self.get_voters_file()
+        for p_uuid in self.p_uuids:
+            upload_voters_location = '/elections/%s/polls/%s/voters/upload' \
+                % (self.e_uuid, p_uuid)
+            self.c.post(
+                upload_voters_location,
+                {'voters_file': file(voter_files[p_uuid]),
+                 'encoding': 'iso-8859-7'}
+                )
+            self.c.post(upload_voters_location, {'confirm_p': 1, 'encoding': 'iso-8859-7'})
+        self.verbose('+ Voters file submitted')
+
+    def voter_login(self, voter, voter_login_url=None):
+        r = self.c.get(voter_login_url, follow=True)
+        assert r.status_code == 200
+
+
+class TestPollViews(TestElectionBase):
+
+    def setUp(self):
+        super(TestElectionBase, self).setUp()
+        self.c.post(self.locations['login'], self.login_data)
+        self.election = self.get_election(name="election", voting_starts_at=datetime.date.today(),
```

Comment:
`get_election` is called only once, why not inline it? I.e. instead of using a separate function, do `self.election = Election.objects.create(...)`